ini.c

```diff
@@ -202,7 +202,12 @@ int ini_parse_stream(ini_reader reader, void* stream, ini_handler handler,
             }
             else if (!error) {
                 /* No '=' or ':' found on name[=:]value line */
-                error = lineno;
+#if INI_ALLOW_KEY_WITHOUT_EQUAL
+                *end = '\0';
+                name = rstrip(start);
+                if (!HANDLER(user, section, name, NULL))
```

Comment:
You need to add a `&& !error` here like the other `HANDLER` calls so that it returns the first error.